staging/src/k8s.io/legacy-cloud-providers/azure/azure.go

```diff
@@ -975,3 +996,12 @@ func (az *Cloud) ShouldNodeExcludedFromLoadBalancer(nodeName string) (bool, erro
 
 	return az.excludeLoadBalancerNodes.Has(nodeName), nil
 }
+
+func isNodeReady(node *v1.Node) bool {
```

Comment:
Curious why you need to check this here because service controller already filters out nodes based on ready condition before passing them into the provider

Reply:
That's true. The thing is, azure-specific code builds its own cache of invalid nodes (`az.excludeLoadBalancerNodes`) using its own informers (https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/legacy-cloud-providers/azure/azure.go#L742-L775), and that's where things weren't handled as expected for nodes that weren't ready (hence this PR!)

The list of available nodes that is passed over by the service controller is now ignored when deciding which nodes to exclude from a load balancer. The code checks against the local cache of invalid nodes and not the list received from the service controller. This PR introduced this change (and the regression I'm fixing here):
https://github.com/kubernetes/kubernetes/pull/105839/files#diff-6000121b1f9282a8bb5c767e3b5f935da6cebdb2c6a0fcd2e5dd36aee1845671L1142-R1138

This was added, I think, in order to possibly add Azure-specific conditions to the whole node inclusion/exclusion mechanism.

Reply:
I also just updated the comment in `reconcileLoadBalancer`, which referred to the old version, prior to https://github.com/kubernetes/kubernetes/pull/105839